hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1967,6 +2035,19 @@ func reconcileHostedControlPlane(hcp *hyperv1.HostedControlPlane, hcluster *hype
 		hcp.Spec.SecretEncryption = hcluster.Spec.SecretEncryption.DeepCopy()
 	}
 
+	// If ROSA passes KubeAPICustomKubeconfig.CustomKubeConfig, we should pass it through to the HCP, otherwise
+	// we will keep it in blank for name auto-generation.
+	// If ROSA passes the KasDNSName but KasDNSName it's not set in the HC, we will not set anything.
+	if len(hcluster.Spec.KasDNSName) >= 0 {
+		if hcp.Spec.KasCustomKubeconfig != nil && hcp.Spec.KasCustomKubeconfig.SecretRef != nil {
```

Comment:
why do we need an if here or the KasCustomKubeconfig struct? seems all the info the hcp controller needs is the KasDNSName

Reply:
same as above

Reply:
To clarify, I've removed the `hcp.spec.KasCustomKubeconfig` because it's an old implementation detail that is not relevant anymore and I've changed the PR to simplify the implementation in that way.